ROSCO/src/Controllers.f90

```diff
@@ -374,6 +383,81 @@ SUBROUTINE YawRateControl(avrSWAP, CntrPar, LocalVar, objInst, zmqVar, DebugVar,
+        TYPE(ObjectInstances), INTENT(INOUT)         :: objInst
+        TYPE(ErrorVariables), INTENT(INOUT)          :: ErrVar
+        TYPE(LidarVariables), INTENT(INOUT)          :: LidVar 
+        INTEGER(IntKi)                               :: ArrayPosition
+        INTEGER(IntKi)                               :: MaxBufferStep
+        REAL(DbKi)                                   :: REWS
+        REAL(DbKi), DIMENSION(:), ALLOCATABLE        :: ActionTime
+             
+       IF (LocalVar%iStatus == 0) THEN  
+             LidVar%REWS_f(1)    = LPFilter(LocalVar%LidSpeed(1), LocalVar%DT, CntrPar%FF_LPFCornerFreq, LocalVar%FP, LocalVar%iStatus, .TRUE., objInst%instLPF)
+             LocalVar%FF_Pitch_Error_old = 0
+             LocalVar%FF_PitchRate = 0
+             LocalVar%PC_PitComFF = 0    
+       ELSE
+       MaxBufferStep = SIZE(LidVar%REWS_f)
+        
+        !--------- Shift the filter and Time in the buffer and assign new values -------------------------
+       LidVar%REWS_f(2:MaxBufferStep) = LidVar%REWS_f(1:MaxBufferStep-1)
+       LidVar%REWS_f_Time(2:MaxBufferStep) = LidVar%REWS_f_Time(1:MaxBufferStep-1)
+       
+        !--------- Determine REWS - This will either average the measurements from the multiple beams or the multiple pulse gates --------
+       IF (LocalVar%SensorType == 1) THEN
+          REWS = SUM(LocalVar%LidSpeed)/LocalVar%NumBeam
+       ELSEIF (LocalVar%SensorType == 2) THEN
+          REWS = SUM(LocalVar%LidSpeed)/1    
+       ELSEIF (LocalVar%SensorType == 3) THEN
+          REWS = SUM(LocalVar%LidSpeed)/LocalVar%NumPulseGate  
+       ENDIF
+ 
+       LidVar%REWS_f(1)    = LPFilter(REWS, LocalVar%DT, CntrPar%FF_LPFCornerFreq, LocalVar%FP, LocalVar%iStatus, .FALSE., objInst%instLPF)
+       LocalVar%LeadTime_1   =  (-LocalVar%MsrPositionsX(1))/LidVar%REWS_f(1)
+       LidVar%REWS_f_time(1) = LocalVar%LeadTime_1 + LocalVar%Time
+           
+        !--------- Get the buffer ID that has the time closest to the pitch action time ------------------
+       ActionTime       =  ABS(LidVar%REWS_f_Time - LocalVar%Time - CntrPar%Pitch_PreviewTime)               ! This determines the time that needs to be matched to the wind speed record
+       ArrayPosition    =  INT(MINLOC(ActionTime, 1))                                                        ! This determines the position  in the REWS array that needs to be called
+       LocalVar%REWS_b  =  LidVar%REWS_f(ArrayPosition)                                                      ! REWS_b is the wind speed AT the desired time
+       LocalVar%FF_Pitch = interp1d(CntrPar%REWS_curve,CntrPar%Pitch_curve, LocalVar%REWS_b, ErrVar)
+       LocalVar%FF_Pitch_Error  = LocalVar%FF_Pitch - (LocalVar%BlPitch(1)+ LocalVar%BlPitch(2)+ LocalVar%BlPitch(3))/3
+       LocalVar%PC_PitComFF =  CntrPar%FF_Kp * LocalVar%FF_Pitch_Error 
+
+       !---------- Limit feedforward pitch error depending on wind speed conditions ----------------------
```

Comment:
This section still looks like it can be generalized into a lookup table.  Would a table of the upper limit and lower limit vs. wind speed work?